diff --git a/chapter-4/postimage.js b/chapter-4/postimage.js
index 988d845..236234d 100644
--- a/chapter-4/postimage.js
+++ b/chapter-4/postimage.js
@@ -10,12 +10,18 @@ import { url } from "./config.js";
 // ファイルをバイナリとして開く
 const testImage = open("testimage.jpg", "b");
 
+// getAccount 関数を accounts.js から import
+import { getAccount } from "./accounts.js";
+
 // k6が実行する関数
 // ログインして画像を投稿するシナリオ
 export default function () {
+  // ランダムに1アカウントを選択
+  const account = getAccount();
+  // /login に対して送信
   const res = http.post(url("/login"), {
-    account_name: "terra",
-    password: "terraterra",
+    account_name: account.account_name,
+    password: account.password,
   });
   const doc = parseHTML(res.body);
   const token = doc.find('input[name="csrf_token"]').first().attr("value");
